modules/sdk-core/src/bitgo/wallet/wallets.ts

```diff
@@ -168,6 +180,58 @@ export class Wallets implements IWallets {
     };
   }
 
+  private async generateLightningWallet(params: GenerateLightningWalletOptions): Promise<LightningWalletWithKeychains> {
+    const reqId = new RequestTracer();
+    this.bitgo.setRequestTracer(reqId);
+
+    const { label, passphrase, enterprise, passcodeEncryptionCode } = params;
+
+    const keychainPromises = ([undefined, 'userAuth', 'nodeAuth'] as const).map((purpose) => {
+      return async (): Promise<Keychain> => {
+        let keychain: KeyPair | null = this.baseCoin.keychains().create();
+        const pub = keychain.pub;
+        const encryptedPrv = this.bitgo.encrypt({ password: passphrase, input: keychain.prv });
+        delete (keychain as any).prv;
+        keychain = null;
+        const keychainParams: AddKeychainOptions = {
+          pub,
+          encryptedPrv,
+          originalPasscodeEncryptionCode: purpose === undefined ? passcodeEncryptionCode : undefined,
+          coinSpecific: purpose === undefined ? undefined : { [this.baseCoin.getChain()]: { purpose } },
+          keyType: 'independent',
+          source: 'user',
+        };
+        return await this.baseCoin.keychains().add(keychainParams);
+      };
+    });
+
+    const { userKeychain, userAuthKeychain, nodeAuthKeychain } = await promiseProps({
+      userKeychain: keychainPromises[0](),
+      userAuthKeychain: keychainPromises[1](),
+      nodeAuthKeychain: keychainPromises[2](),
+    });
+
+    const walletParams: SupplementGenerateWalletOptions = {
+      label,
+      m: 1,
+      n: 1,
+      type: 'hot',
+      enterprise,
+      keys: [userKeychain.id],
+      coinSpecific: { [this.baseCoin.getChain()]: { keys: [userAuthKeychain.id, nodeAuthKeychain.id] } },
+    };
+
+    const newWallet = await this.bitgo.post(this.baseCoin.url('/wallet')).send(walletParams).result();
+    const wallet = new Wallet(this.bitgo, this.baseCoin, newWallet);
+    return {
+      wallet,
+      userKeychain,
+      userAuthKeychain,
+      nodeAuthKeychain,
```

Comment:
```suggestion
      userKeychain,
      userAuthKeychain,
      nodeAuthKeychain,
```

these only exist for the LN case so I don't think the `walletType` is required